File: scratch/search_history.js
const fs = require('fs');
const readline = require('readline');
const path = require('path');

const logPath = 'C:\\Users\\info\\.gemini\\antigravity-ide\\brain\\95152b10-828b-4808-98d6-f9529b177cf6\\.system_generated\\logs\\transcript.jsonl';

if (!fs.existsSync(logPath)) {
  console.error('Log file not found:', logPath);
  process.exit(1);
}

const rl = readline.createInterface({
  input: fs.createReadStream(logPath),
  output: process.stdout,
  terminal: false
});

let found = false;
rl.on('line', (line) => {
  if (line.includes('fetchAssemblyData')) {
    found = true;
    console.log('--- FOUND MATCHING LINE ---');
    // 長いので一部だけ切り出して表示
    const idx = line.indexOf('fetchAssemblyData');
    console.log(line.substring(Math.max(0, idx - 100), Math.min(line.length, idx + 1000)));
  }
});

rl.on('close', () => {
  if (!found) {
    console.log('No matches found for fetchAssemblyData in transcript.jsonl');
  }
});

File: scratch/search_history.py
import json
import re

log_path = r"C:\Users\info\.gemini\antigravity\brain\1577f144-a8f1-437e-a57d-752724db685f\.system_generated\logs\transcript.jsonl"

with open(log_path, "r", encoding="utf-8") as f:
    for line in f:
        try:
            data = json.loads(line)
            # settingsTab === 'fee' を含む tool_calls や content を検索
            str_data = json.dumps(data, ensure_ascii=False)
            if "settingsTab === 'fee'" in str_data or "settingsTab === 'stripe'" in str_data:
                print(f"--- Step {data.get('step_index')} (Type: {data.get('type')}) ---")
                # tool_calls があればその引数を表示
                if 'tool_calls' in data:
                    for tc in data['tool_calls']:
                        if tc['name'] in ['replace_file_content', 'write_to_file', 'multi_replace_file_content']:
                            print(f"Tool Call: {tc['name']}")
                            args = tc.get('args', {})
                            # 文字列に変換して一部分だけ出す
                            args_str = json.dumps(args, ensure_ascii=False)
                            print(args_str[:1000] + "...")
                # content に含まれていれば一部分を表示
                content = data.get('content', '')
                if content:
                    print("Content preview:")
                    print(content[:500] + "...")
        except Exception as e:
            print("Error parsing line:", e)
